app/controllers/ops_controller/diagnostics.rb

```diff
@@ -350,6 +347,41 @@ def diagnostics_server_list
 
   private ############################
 
+  def fetch_local_log(*params)
+    if MiqEnvironment::Command.supports_systemd?
+      fetch_journal_log(params)
+    elsif !MiqEnvironment::Command.is_podified?
+      fetch_log_file(params)
+    end
+  end
+
+  def fetch_journal_log(service_name = nil, max_count = 1_000, filter_params = {})
+    return unless MiqEnvironment::Command.supports_systemd?
+
+    filter_params[:syslog_identifier] = service_name if service_name.present?
+
+    require "systemd/journal"
+    Systemd::Journal.open do |journal|
+      journal.filter(filter_params)
+
+      entries = max_count.nil? ? journal.all : journal.take(max_count)
```

Comment:
If we aren't given a `max_count` we might still want to limit it to either the current boot or if I can detect if we're past the current journal file (the latter being equivalent to reading all of `evm.log` but not including the rotated log files)